src/components/scrollManager.js

```diff
@@ -0,0 +1,501 @@
+define(["dom", "browser", "layoutManager"], function (dom, browser, layoutManager) {
+    "use strict";
+
+    /**
+     * Scroll time in ms.
+     */
+    var ScrollTime = 270;
+
+    /**
+     * Epsilon for comparing values.
+     */
+    var Epsilon = 1e-6;
+
+    // FIXME: Need to scroll to top of page to fully show the top menu. This can be solved by some marker of top most elements or their containers
+    var _minimumScrollY = 0;
+    /**
+     * Returns minimum vertical scroll.
+     * Scroll less than that value will be zeroed.
+     *
+     * @return {number} minimum vertical scroll
+     */
+    function minimumScrollY() {
+        if (_minimumScrollY === 0) {
+            var topMenu = document.querySelector(".headerTop");
+            if (topMenu) {
+                _minimumScrollY = topMenu.clientHeight;
```

Comment:
what happens if `.headerTop` ever changes size? cannot we just never cache it and always return `topMenu.clientHeight` if it's available

Reply:
I tried to minimize the code that needs to be executed. I worry about `querySelector`. But maybe I over-optimized it :smile: , because this function is called only once on each user navigation.

Reply:
Yeah, this looks like premature optimization :) I think you shouldn't cache stuff here. Also that selector should be pretty easy to query for...